include/aspect/particle/integrator/euler.h

```diff
@@ -0,0 +1,68 @@
+/*
+ Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_integrator_euler_h
+#define __aspect__particle_integrator_euler_h
+
+#include <aspect/particle/integrator/interface.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Integrator
+    {
+      /**
+       * Euler scheme integrator, where y_{n+1} = y_n + dt * v(y_n).
+       * This requires only one step per integration, and doesn't involve any extra data.
+       */
+      template <int dim>
+      class EulerIntegrator : public Interface<dim>
+      {
+        public:
+          /**
+           * Perform an integration step of moving the particles of one cell
+           * by the specified timestep dt. Implementations of this function
+           * must update the particle location. Between calls to this function
+           * the velocity at the updated particle positions is evaluated and
+           * passed to integrate_step during the next call.
+           *
+           * @param [in] begin_particle An iterator to the first particle to be moved.
+           * @param [in] end_particle An iterator to the last particle to be moved.
+           * @param [in] old_velocities The velocities at t_n, i.e. before the
+           * particle movement.
+           * @param [in] velocities The velocities at t_{n+1}, i.e. after the
+           * particle movement.
```

Comment:
I think this is confusing(ly worded). What you probably want to say is that it is the velocities at the location of the particles *at the end of the time step from $t_n$ to $t_{n+1}$*. The fact that the particles move has no effect on these velocities, and they are not evaluated at the *new* locations of the particles either.